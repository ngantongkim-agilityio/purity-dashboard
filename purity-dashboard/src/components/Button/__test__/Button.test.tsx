// Libs

import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Components
import { Button } from '..';

describe('Button component', () => {
  it('Should render Button component with default props', () => {
    const { container } = render(<Button>Click</Button>);

    expect(container).toMatchSnapshot();
  });

  it('Simulate press event and expect mock function to be called', () => {
    const handlePress = jest.fn();
    const { getByTestId } = render(
      <Button data-testid='test-button' onPress={handlePress} isLoading>
        Click
      </Button>
    );
    const button = getByTestId('test-button');

    userEvent.click(button);

    waitFor(() => {
      expect(handlePress).toHaveBeenCalled();
    });
  });

  it('Should be disabled when the disabled prop is true', () => {
    render(<Button disabled>Press One</Button>);

    expect(screen.getByText('Press One')).toBeDisabled();
  });

  test('Should not call click function when isLoading prop is true', () => {
    const handleClick = jest.fn();

    const { getByTestId } = render(
      <Button data-testid='test-button' onClick={handleClick} isLoading>
        Click
      </Button>
    );
    const button = getByTestId('test-button');
    fireEvent.click(button);

    waitFor(() => {
      expect(handleClick).not.toHaveBeenCalled();
    });
  });
});
