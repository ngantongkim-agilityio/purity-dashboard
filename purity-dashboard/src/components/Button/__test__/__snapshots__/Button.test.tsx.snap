// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Should render Button component with default props 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent font-semibold p bg-primary uppercase text-white px-4 min-w-20 h-9 text-xs gap-2 rounded-xl"
    type="button"
  >
    Click
  </button>
</div>
`;

exports[`Button component should render Button component with default props 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 min-w-20 gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent font-semibold bg-primary-100 disabled:bg-background-400 text-primary-500 py-10 px-20 h-21 text-xs"
    type="button"
  >
    Click me
  </button>
</div>
`;
